metadata-ingestion/src/datahub/utilities/parsing_partition_util.py

```diff
@@ -0,0 +1,29 @@
+import datetime
+import logging
+from typing import Tuple
+
+from dateutil.relativedelta import relativedelta
+
+logger = logging.getLogger(__name__)
+
+
+def get_partition_range_from_partition_id(
+    partition_id: str, partition_datetime: datetime.datetime
+) -> Tuple[datetime.datetime, datetime.datetime]:
+    duration: relativedelta
+    # if yearly partitioned,
+    if len(partition_id) == 4:
```

Comment:
Can we put the date parsing here as well? 
like:
```
partition_datetime = datetime.strptime(dt_string, "%Y%m%d")
```

Currently, I think the partition with the hour part can't be parsed because the parser detects wrongly the DateTime.
With specific formats we can resolve that issue as well.

Reply:
I see. I will make a change. @treff7es

Reply:
If this is merged, https://github.com/datahub-project/datahub/pull/4591 this pull request is not necessary. Once this PR is merged, I will close 4591 PR. @treff7es